### ISSUE TITLE
`restore-file` occasionally restores the wrong commit

### ISSUE THREAD

**Issue by fregante** [refined-github/refined-github#4679]
### Please ensure:

- [X] The bug is caused by Refined GitHub. It doesn't happen if I disable the extension.

### Description

Rare occurrence but:

1. Create `X` branch from `main`
2. Force-push `main` to something else
3. Create PR from branch `X`

Now you'll have a PR with the changes that the force-push made.

4. Click "Restore file" on one of the files in the PR

You'll notice that the feature "succeeds" but the commits are empty. This is a live example: [`419c37c` (#4677)](https://github.com/sindresorhus/refined-github/pull/4677/commits/419c37c29ed4f4961d48495fadc951140c249f99)

### Example URL

https://github.com/sindresorhus/refined-github/pull/4677, before it was rebased

### Browser(s) used

Safari

**Comment by fregante** [comment]
Maybe it's not even due to force pushes. There were no force pushes in https://github.com/npmhub/npmhub/pull/141, but this "Restore" commit is incorrect: [`7485182` (#141)](https://github.com/npmhub/npmhub/pull/141/commits/74851825ea433e27f411eb1733d3c6cdddd3c82c)

The issue specifically is that it "correctly" fetched the file from the base branch (`main`), but GitHub still shows the file as a "change" in the PR (even though when merged nothing will actually change): https://github.com/npmhub/npmhub/pull/141/files#diff-719f723e7f36f1f0192a75b8750569dcb1319113ea7e038109894895d2ea7d74

You can try restoring the same file multiple times but that's what you get.

In short, I think we should fetch the file not from the base _branch_ but from the base _commit_.

I think the repro is:

1. Create PR with file X
2. Change the same file X from the PR directly on the base branch
3. Restore file X in the PR

**Comment by fregante** [comment]
"merge base" is what we're looking for: https://stackoverflow.com/q/1549146/288906

v3 seems to have `merge_base_commit`, which is likely what we want: https://docs.github.com/en/rest/commits/commits?apiVersion=2022-11-28#compare-two-commits

It seems that it also exists on v4 as `baseTarget`, I'll try it

```gql
{
  repository(owner: "refined-github", name: "sandbox") {
    pullRequest(number: 68) {
      baseRefOid
      mergeable
      viewerCanEditFiles
      headRef {
        compare(headRef: "default-a") {
          status
          aheadBy
          baseTarget {
            oid
          }
        }
      }
    }
  }
}
```
